src/preamble.js

```diff
@@ -978,12 +978,33 @@ function createWasm(env) {
     return value;
   };
 #endif
+var proxyHandler = {
+  'get': function(obj, prop) {
+    if (prop in obj) {
+      return obj[prop]; // already present
+    }
+
+    // Check if the minification mapping exists first, if not just
+    // use the original name.
+    var funcName = prop;
+    if (Module["mapping"] && Module["mapping"][prop]) {
+      Module["mapping"][prop];
+    }
+    
+    // For a missing property generate a stub that will do a runtime lookup and error out if its still missing
+    return env[prop] = function() {
+#if ASSERTIONS
+      assert(Module[funcName], 'missing linked function ' + funcName + '. perhaps a side module was not linked in? if this function was expected to arrive from a system library, try to build the MAIN_MODULE with EMCC_FORCE_STDLIBS=1 in the environment');
+#endif
+      return Module[funcName].apply(null, arguments);
+    };
```

Comment:
It looks like this change turns startup errors into lazy errors later on when actually called - that is only needed with dynamic linking, correct? So we can put it behind `MAIN_MODULE` I think.

Reply:
Sure, will add this in.